documents/description.md

```diff
@@ -2,8 +2,35 @@
+
+Students at the Claremont Colleges struggle with these same questions. We have some tools to help us visualize the possibilities for the subsequent semester, such as the [Scheduler App](http://scheduler.5capps.com/#) and the [ASPC-Scheduler App](https://aspc.pomona.edu/courses/schedule/), but neither seems to give the users the full flexibility they may want and need to specify exactly what they are looking for. A DSL such as CourseScheduler, which I am designing will allow students to express their preferences for courses with the fluidity and expressiveness of a full language. Rather than looking through all the possible schedules with different combinations of the classes you are interested in, students should be able to input a list of all the courses they might be interested in, and then write a program describing the kind of schedule they want, run it, and get a list of the possibly schedules that most closely align with their preferences.
+
 ## Language domain
 
+The domain of the language is classes and class schedules, which is a useful domain in so far as it is applicable to all students trying to plan their schedule based on a set of classes they have varying degrees of interest in. My language will be designed specifically based on classes at the 5Cs and may contain some things specific to classes offered here, such as an interface with the 5C portal, but it should be possible to generalize it to meet the needs of students at other schools as well. The [Scheduler App](http://scheduler.5capps.com/#) and the [ASPC-Scheduler App](https://aspc.pomona.edu/courses/schedule/), are softwares that have already been developed in this domain to assist students, and I intend to look at the way the Scheduler App gets information about classes from portal and borrow from that, with the permission of the student who developed it. However, neither app progrives the full flexibility and expressiveness of a language that would allow a student to fully express their preferences and I want to develope a language that allows that. 
+
 ## Language design
 
+In one sentence, my DSL design would be: Constraint satisfaction generating course schedules. A program in my language will be a set of classes input to the system which may need to be stored in some type of data structure and then a list of constraints the user wants to impose on the output list of schedules along with a specification of the number of possible schedules the user would like to have generated. The output should be a list of several possible schedules that align as closely as possible with the users stated preferences. In the future, the output might be transformed into a graphical output like the Scheduler App, so that the user can better visualize a particular schedule.
```

Comment:
It might be helpful to distinguish inputs from the program itself (which is not always a clear distinction). I would say that, perhaps, the input is the course data and the _program itself_ is the list of constraints.